homeassistant/components/melissa/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow for Melissa integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import melissa
+from melissa.exceptions import ApiException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+class PlaceholderHub:
+    """Placeholder class to make tests pass."""
+
+    def __init__(self) -> None:
+        """Initialize."""
+
+    async def authenticate(self, username: str, password: str) -> bool:
+        """Test if we can authenticate with the host."""
+        api = melissa.AsyncMelissa(username=username, password=password)
+        try:
+            await api.async_connect()
+        except ApiException:
+            _LOGGER.error("Wrong Username/Password")
+
+        connected = api.refresh_token is not None
+        if api.session is not None:
+            await api.session.close()
+        return connected
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+    hub = PlaceholderHub()
+
+    if not await hub.authenticate(data["username"], data["password"]):
+        raise InvalidAuth
+
+    # Return info that you want to store in the config entry.
+    return {"title": "MClimate"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Melissa."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_import(self, user_input):
+        """Handle import."""
+        return await self.async_step_user(user_input)
```

Comment:
Thanks for adding the import!

```suggestion
        """Handle import."""
        async_create_issue(
            self.hass,
            DOMAIN,
            "deprecated_yaml",
            breaks_in_ha_version="2023.4.0",
            is_fixable=False,
            severity=IssueSeverity.WARNING,
            translation_key="deprecated_yaml",
        )
        return await self.async_step_user(user_input)
```
Please also add a issue for warning the user that the YAML config is obsolete.

Another thing is that we need a test for the `config_flow` (see [config_entries_config_flow_handler](https://developers.home-assistant.io/docs/config_entries_config_flow_handler#testing-your-config-flow).